### PR TITLE
ADO_181037: Docker images upgraded

### PR DESCRIPTION
**Pull request by rsimone**
I pulled all the docker images and checked them using the docker scout tool:

docker scout cves < docker image >

I found the libwebp vulnerability in the navionics/ruby:2.3.8v3 and navionics/ruby:2.7.4v2
Building the ruby images again with no changes the vulnerability is not present anymore.
I uploaded the new version on dockerhub (navionics/ruby:2.3.8v4 and navionics/ruby:2.7.4v3)

I updated the url_base in maven 3.8.7 dockerfile because the previous one is not available anymore.
I updated the  [ruby/Dockerfile.ruby2.3.8](https://github.com/Navionics/Dockers/pull/23/files#diff-1c5513e28f70aa71d39683046015d735454d87618b830cbe2203a36b4c2269fb) docker images because to fix a problem with the gem ffi version.

### CODE REVIEW

**Review by agentili — DISMISSED** [review]

**Review by gianninisimone — APPROVED** [review]

File: maven/3.8.8/openjdk17/Dockerfile [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-ARG SHA=21c2be0a180a326353e8f6d12289f74bc7cd53080305f05358936f3a1b6dd4d91203f4cc799e81761cf5c53c5bbe9dcc13bdb27ec8f57ecf21b2f9ceec3c8d27
+ARG SHA=332088670d14fa9ff346e6858ca0acca304666596fec86eea89253bd496d3c90deae2be5091be199f48e09d46cec817c6419d5161fb4ee37871503f472765d00
 ARG BASE_URL=https://downloads.apache.org/maven/maven-3/${MAVEN_VERSION}/binaries
 
 ENV MAVEN_HOME /usr/share/maven
 ENV MAVEN_CONFIG "$USER_HOME_DIR/.m2"
 
 RUN set -x \
   && apt-get update \
+  && apt-get upgrade -y \
```
Comment (mcnavionics) [inline maven/3.8.8/openjdk17/Dockerfile:13]: how about re-releasing a fresher and bug-free navionics/openjdk:17-debian instead? and build this on top of it.

File: cpp/Dockerfile.cppcheck.ubuntu.20.04 [hunk 2]
```diff
@@ -4,6 +4,7 @@ ENV DEBIAN_FRONTEND=noninteractive
 ENV TERM linux
 
 RUN apt-get update && \
+  apt-get upgrade -y && \
```
[the remark this answers is not in the document]
Comment (mcnavionics) [reply cpp/Dockerfile.cppcheck.ubuntu.20.04:7]: more occurrences in the PR omitted.

[the remark this answers is not in the document]
Comment (rsimone) [reply cpp/Dockerfile.cppcheck.ubuntu.20.04:7]: @mcnavionics  So we need only to build the image again (with no docker changes) and tag it?

File: maven/3.8.8/openjdk17/Dockerfile [hunk 3]
```diff
@@ -38,4 +38,4 @@ RUN chmod +x /usr/local/bin/mvn-entrypoint.sh
 COPY ./maven/${MAVEN_VERSION}/openjdk17/settings-docker.xml /usr/share/maven/ref/
 
 ENTRYPOINT ["/usr/local/bin/mvn-entrypoint.sh"]
-CMD ["mvn"]
\ No newline at end of file
+CMD ["mvn"]
```
Comment (mcnavionics) [inline maven/3.8.8/openjdk17/Dockerfile:41]: can new line change be avoided?

File: ruby/Dockerfile.ruby2.3.8 [hunk 4]
```diff
@@ -35,6 +35,7 @@ RUN curl -L http://cache.ruby-lang.org/pub/ruby/2.3/ruby-2.3.8.tar.gz > ruby-2.3
 # Install base gems
 RUN gem install bundler -v 1.17.3 \
     && gem install rubygems-update -v 3.3.26 --no-document \
+    && gem install ffi -v 1.15.5 \
```
Comment (mcnavionics) [inline ruby/Dockerfile.ruby2.3.8:38]: why is this needed?

File: maven/3.8.8/openjdk17/Dockerfile [hunk 5]
```diff
@@ -1,8 +1,8 @@
 FROM navionics/openjdk:17-debian
 
-ARG MAVEN_VERSION=3.8.7
+ARG MAVEN_VERSION=3.8.8
```
Comment (mcnavionics) [inline maven/3.8.8/openjdk17/Dockerfile:3]: I would keep maven version update out of scope, even if minor.

Comment (mcnavionics) [reply maven/3.8.8/openjdk17/Dockerfile:3]: Is this because image uses only maven version for its versioning?

**Review by agentili — DISMISSED** [review]

File: ruby/Dockerfile.ruby2.3.8 (hunk 4 above)
Comment (rsimone) [reply ruby/Dockerfile.ruby2.3.8:38]: This is need to fix the version of ffi gem. rb-inotify requires ffi: if the version is not fixed the gem tool try to install the last version of ffi and it fails because it requires ruby version > 2.5

File: maven/3.8.8/openjdk17/Dockerfile (hunk 3 above)
Comment (rsimone) [reply maven/3.8.8/openjdk17/Dockerfile:41]: very strange: no new line present in the file https://github.com/Navionics/Dockers/blob/ADO_181037_webp_vulnerability/maven/3.8.8/openjdk17/Dockerfile

File: maven/3.8.8/openjdk17/Dockerfile (hunk 5 above)
Comment (mcnavionics) [reply maven/3.8.8/openjdk17/Dockerfile:3]: It is available from the archived downloads https://archive.apache.org/dist/maven/maven-3/3.8.7/binaries/
Minor version update should not harm though.

File: ruby/Dockerfile.ruby2.3.8 (hunk 4 above)
Comment (mcnavionics) [reply ruby/Dockerfile.ruby2.3.8:38]: Was it working before? What changed?

File: maven/3.8.8/openjdk17/Dockerfile [hunk 6]
```diff
@@ -1,8 +1,8 @@
...
 ARG USER_HOME_DIR="/root"
-ARG SHA=21c2be0a180a326353e8f6d12289f74bc7cd53080305f05358936f3a1b6dd4d91203f4cc799e81761cf5c53c5bbe9dcc13bdb27ec8f57ecf21b2f9ceec3c8d27
+ARG SHA=332088670d14fa9ff346e6858ca0acca304666596fec86eea89253bd496d3c90deae2be5091be199f48e09d46cec817c6419d5161fb4ee37871503f472765d00
 ARG BASE_URL=https://downloads.apache.org/maven/maven-3/${MAVEN_VERSION}/binaries
```
Comment (mcnavionics) [inline maven/3.8.8/openjdk17/Dockerfile:6]: Using archived downloads is more reliable for reproducible docker image builds, eg: https://archive.apache.org/dist/maven/maven-3/3.8.8/binaries/

It would also work for previous 3.8.7, without changing this file, which is out of scope.

File: ruby/Dockerfile.ruby2.3.8 (hunk 4 above)
Comment (rsimone) [reply ruby/Dockerfile.ruby2.3.8:38]: Before it worked probably because the last version of gem ffi did not exist

File: maven/3.8.8/openjdk17/Dockerfile (hunk 6 above)
Comment (rsimone) [reply maven/3.8.8/openjdk17/Dockerfile]: Ok. I updated the url_base in the 3.8.7 version.

**Review by aielloaldo — APPROVED** [review]
